### PR TITLE
Remove question mark from details URLs

### LINKED ISSUE #694: [Bug] Remove question mark from URL in v2
- [x] I'd be willing to implement a fix

**Describe the bug**

URLs now are like: https://next.yarnpkg.com/package/?@wavevision/class-name

in V1 the URL is https://next.yarnpkg.com/package/@wavevision/class-name

This breaks file URLs, since they assumed the query string wasn't set yet: https://next.yarnpkg.com/package?%40wavevision%2Fclass-name&files

**To Reproduce**

Go to any package detail URL, see the extra ?
**Additional context**

Any change from v1 to v2 will be a _Breaking_ change if the URL is changed, so this needs to be fixed :)

### PR DESCRIPTION
**Pull request by Haroenv**
I haven't tried to run this on my machine yet, so let's make sure we wait for the deploy preview before merging.

**What's the problem this PR addresses?**

fixes #694

**How did you fix it?**

Strategy comes from: https://github.com/yarnpkg/website/blob/8f2614827ac325c212d8dfa823d82bda897c9df0/js/src/package.js#L12-L20

### CODE REVIEW

**Comment by Haroenv** [comment]
hmm, I can't see why that error exists in the build, since I don't use .slice, do you know @arcanis ?

File: packages/gatsby/src/pages/package.js [hunk 1]
```diff
@@ -41,7 +41,12 @@ const PackagePage = ({ searchState, onSearchStateChange }) => {
   const [tags, setTags] = useState([]);
   const [owners, setOwners] = useState([]);
 
-  const packageName = typeof window !== 'undefined' && window.location.search.slice(1);
+  const [
+    /* leading slash */,
+    /* package/ */,
+    ...parts
+  ] = typeof window !== 'undefined' && window.location.pathname.split('/');
```
Comment (arcanis) [inline packages/gatsby/src/pages/package.js]: Your slice likely comes from here, after the code has been transpiled. When running SSR, you'll be destructuring `undefined`.

**Comment by Haroenv** [comment]
The router still needs to be set up to deal with the /packages/*

**Comment by Haroenv** [comment]
hmm, it still seems to flash the 404 (meaning the server side rendering didn't work properly. I might need to add the Router)

**Comment by arcanis** [comment]
Still an improvement! Merging, and we can keep iterating 😃

**Pull request merged**